### DISCUSSION TITLE
the system detected an invalid pointer address osu-yes i did look for solutions, it did not work.

### DISCUSSION THREAD

**Discussion by khaniare** [ppy/osu#29908] [Q&A] [answered]
Today while opening up osu, i was confused, but i was logged out, i tried logging back in, but got the error "the system detected an invalid pointer address." or something across the lines of invalid pointer address.

searched it up and got this post https://github.com/ppy/osu/discussions/28158, thing is the solution (to put netsh winsock reset), did **not** work. yes i did restart afterwards.)

i tried multiple times to open up osu again and try logging in, still didn't work. osu was working fine before, just took a break for around a day or two, and come back to this. logs are down below. hopefully there's a fix. wonder what's wrong.

[1726616182.auth.log](https://github.com/user-attachments/files/17036422/1726616182.auth.log)
[1726616182.nauth.log](https://github.com/user-attachments/files/17036423/1726616182.nauth.log)
[1726616182.network.log](https://github.com/user-attachments/files/17036425/1726616182.network.log)
[1726616182.performance.log](https://github.com/user-attachments/files/17036426/1726616182.performance.log)
[1726616182.runtime.log](https://github.com/user-attachments/files/17036427/1726616182.runtime.log)

**Comment by khaniare** [comment]
**Update**: have also tried the following, still getting the same error. 

- `netsh int ip reset`
- `netsh interface ipv4 reset`
- `netsh interface ipv6 reset`
- also tried flushing the DNS cache, I ran: `ipconfig /flushdns`

**Comment by khaniare** [comment]
**Update_2**: have done the big nuke.. using my system restore point. apparently the problem was using the opensourced project [TerabyteTweaker](https://github.com/Teramanbr/TerabyteTweaker) (and specifically changing the network settings. 

(if you want to find out what i mean, download the file, press c, and you can see the network option, that's what the problem was.) 

this is the specific part of code that messed it all up... is there any way to fix this?

'if "%internetgame%"=="false" goto skipinternet
echo Internet Cache Cleaning
ipconfig /release >nul 2>&1
ipconfig /renew >nul 2>&1
arp -d * >nul 2>&1
nbtstat -R >nul 2>&1
nbtstat -RR >nul 2>&1
ipconfig /flushdns >nul 2>&1
ipconfig /registerdns >nul 2>&1
netsh int tcp set global autotuninglevel=normal >nul 2>&1
netsh interface 6to4 set state disabled >nul 2>&1
netsh int isatap set state disable >nul 2>&1
netsh int tcp set global timestamps=disabled >nul 2>&1
netsh int tcp set heuristics disabled >nul 2>&1
netsh int tcp set global chimney=disabled >nul 2>&1
netsh int tcp set global ecncapability=disabled >nul 2>&1
netsh int tcp set global rsc=enabled >nul 2>&1
netsh int tcp set global nonsackrttresiliency=disabled >nul 2>&1
netsh int tcp set security mpp=disabled >nul 2>&1
netsh int tcp set security profiles=disabled >nul 2>&1
netsh int ip set global icmpredirects=disabled >nul 2>&1
netsh int tcp set security mpp=disabled profiles=disabled >nul 2>&1
netsh int ip set global multicastforwarding=disabled >nul 2>&1
netsh int tcp set supplemental internet congestionprovider=ctcp >nul 2>&1
netsh interface teredo set state disabled >nul 2>&1
netsh winsock reset >nul 2>&1
netsh int isatap set state disable >nul 2>&1
netsh int ip set global taskoffload=disabled >nul 2>&1
netsh int ip set global neighborcachelimit=4096 >nul 2>&1
netsh int tcp set global dca=enabled >nul 2>&1
netsh int tcp set global netdma=enabled >nul 2>&1
PowerShell Disable-NetAdapterLso -Name "*" >nul 2>&1

echo Activating MTU Tweaks + Google DNS
set MTU=1473
set LASTGOOD=0
set LASTBAD=65536
set PACKETSIZE=28
set SERVER=google.com

::Check server reachability.
ping -n 1 -l 0 -f -4 google.com >nul 2>&1
if !ERRORLEVEL! NEQ 0 (
  goto :error
)
::Start looking for the maximum MTU.
:seek
ping -n 1 -l !MTU! -f -4 !SERVER! 1>nul
if !ERRORLEVEL! EQU 0 (
  set /A LASTGOOD=!MTU!
  set /A "MTU=(!MTU! + !LASTBAD!) / 2"
  if !MTU! NEQ !LASTGOOD! goto :seek
) else (
  set /A LASTBAD=!MTU!  
  set /A "MTU=(!MTU! + !LASTGOOD!) / 2"
  if !MTU! NEQ !LASTBAD! goto :seek
)
rem Print the result.
set /A "MAXMTU=!LASTGOOD! + !PACKETSIZE!"
rem Export %MAXMTU% variable.
set MAXMTU=%MAXMTU%
goto MTUWorks

:error
rem When something unexpected occurs.
set MAXMTU=-1

:MTUworks
wmic nicconfig where (IPEnabled=TRUE) call SetDNSServerSearchOrder ("8.8.8.8", "8.8.4.4") >nul 2>&1
Reg.exe add "HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\NlaSvc\Parameters\Internet" /v "ActiveDnsProbeContent" /t REG_SZ /d "8.8.4.4" /f >nul 2>&1
Reg.exe add "HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\NlaSvc\Parameters\Internet" /v "ActiveDnsProbeContentV6" /t REG_SZ /d "2001:4860:4860::8844" /f >nul 2>&1
Reg.exe add "HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\NlaSvc\Parameters\Internet" /v "ActiveDnsProbeHost" /t REG_SZ /d "dns.google" /f >nul 2>&1
Reg.exe add "HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\NlaSvc\Parameters\Internet" /v "ActiveDnsProbeHostV6" /t REG_SZ /d "dns.google" /f >nul 2>&1
Reg.exe add "HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\NlaSvc\Parameters\Internet" /v "ActiveWebProbeHost" /t REG_SZ /d "www.msftconnecttest.com" /f >nul 2>&1
Reg.exe add "HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Services\NlaSvc\Parameters\Internet" /v "EnableActiveProbing" /t REG_DWORD /d "1" /f >nul 2>&1

:skipinternet2
if "%internetgame%"=="false" echo Skipping MTU+DNS Tweaks
ping 127.0.0.1 -n 3 >nul 2>&1
set/a progress=%progress% +1
goto Loading'


    tried scraping the code so it isn't that long, but kept most of it to show what the code actually does (do note; this problem does still exist, i fixed it by using a system restore point, but anyone who uses this, will get the same error. hopefully there is an actual fix to this. if not, i'll close it soon!)

**Reply by peppy** [reply, marked as answer]
change system settings, break stuff.

maybe avoid this.
